### PR TITLE
Autofill Names on forms

### LINKED ISSUE #323: Autofill Names on forms
- [ ] Autofil Name of sampler from login details when filling in sample collection form
- [ ] Autofill sender name and if available, sender contact information, on sample shipment form

### PR DESCRIPTION
**Pull request by syedowaisali90**
Hi, @vincent-karuri I have created a PR for review.
Close #323

### CODE REVIEW

**Comment by vincent-karuri** [comment]
The prepopulation should be done on both the sample form and sample shipment form https://github.com/onaio/rdt-standard/issues/323#issue-662533312.

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java [hunk 1]
```diff
@@ -85,6 +87,12 @@ private void  prePopulateSampleCollectionFormFields(JSONObject field, String uni
         if (CovidConstants.FormFields.PATIENT_INFO_DOB.equals(field.getString(KEY))) {
             fillPatientData(field, "Patient Dob: ", patient.getDob());
         }
+
+        // pre-populate the sampler name
+        if (CovidConstants.FormFields.SAMPLER_NAME.equals(field.getString(KEY))) {
+            final AllSharedPreferences allSharedPreference = RDTApplication.getInstance().getContext().allSharedPreferences();
+            field.put(JsonFormConstants.VALUE, allSharedPreference.getANMPreferredName(allSharedPreference.fetchRegisteredANM()));
```
Comment (vincent-karuri) [inline rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java:94]: I would move this into a private method to be shared by both the sample collection and sample shipment form.

File: rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtilsTest.java [hunk 2]
```diff
@@ -822,6 +822,10 @@ else if (CovidConstants.FormFields.PATIENT_INFO_DOB.equals(field.getString(JsonF
                 numOfPopulatedFields++;
             }
         }
+        else if (CovidConstants.FormFields.SAMPLER_NAME.equals(field.getString(JsonFormConstants.KEY))) {
+            assertEquals(field.getString(VALUE), "");
```
Comment (vincent-karuri) [inline rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtilsTest.java:826]: We should assert that this value is actually pre-populated with a non-empty String.

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java (hunk 1 above)
Comment (vincent-karuri) [reply rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java:94]: I also think we should use `JsonFormConstants.TEXT` here, verify that it works with `JsonFormConstants.VALUE` and if so you can leave it as is.

Comment (syedowaisali90) [reply rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java]: fixed.

File: rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtilsTest.java (hunk 2 above)
Comment (syedowaisali90) [reply rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtilsTest.java]: fixed.

File: rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtilsTest.java [hunk 3]
```diff
@@ -822,6 +834,10 @@ else if (CovidConstants.FormFields.PATIENT_INFO_DOB.equals(field.getString(JsonF
                 numOfPopulatedFields++;
             }
         }
+        else if (CovidConstants.FormFields.SAMPLER_NAME.equals(field.getString(JsonFormConstants.KEY))) {
+            assertEquals(field.getString(VALUE), ANM_PREFERRED_NAME);
+            numOfPopulatedFields++;
+        }
```
Comment (vincent-karuri) [inline rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtilsTest.java:840]: We'll need a test for the sample shipment field as well.

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java [hunk 4]
```diff
@@ -121,4 +131,17 @@ private void fillPatientData(JSONObject field, String prefix, String value) thro
...
+
+    private void prePopulateSampleShipmentDetailsFormFields(JSONObject field) throws JSONException {
+
+        // pre-populate the sender name
+        if (CovidConstants.FormFields.SENDER_NAME.equals(field.getString(KEY))) {
+            field.put(JsonFormConstants.VALUE, getNameOfSampler());
+        }
+    }
+
+    private String getNameOfSampler() {
```
Comment (vincent-karuri) [inline rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java:143]: Please rename this to `getLoggedInUserPrefferredName`.

File: rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtilsTest.java (hunk 3 above)
Comment (syedowaisali90) [reply rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtilsTest.java:843]: fixed.

File: rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java (hunk 4 above)
Comment (syedowaisali90) [reply rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/util/CovidRDTJsonFormUtils.java]: fixed.

**Review by vincent-karuri — APPROVED** [review]

**Pull request merged**